sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala

```diff
@@ -194,7 +243,11 @@ object RewriteDistinctAggregates extends Rule[LogicalPlan] {
       val regularAggOperatorMap = regularAggExprs.map { e =>
         // Perform the actual aggregation in the initial aggregate.
         val af = patchAggregateFunctionChildren(e.aggregateFunction)(regularAggChildAttrLookup.get)
-        val operator = Alias(e.copy(aggregateFunction = af), e.sql)()
+        val filterOpt = e.filter.map { fe =>
+          val newChildren = fe.children.map(c => regularAggChildAttrLookup.getOrElse(c, c))
```

Comment:
if we do it correctly as https://github.com/apache/spark/pull/26656/files#r359228858 , we don't need this change.

Reply:
Let me try it.